File: models/OrdemServico.py
from db import db


class OrdemServico(db.Model):
    __tablename__ = 'ordem_servico'

    id = db.Column(db.Integer, primary_key=True)
    cliente_id = db.Column(db.Integer, db.ForeignKey(
        'Clientes.id'), nullable=True)
    usuario_id = db.Column(db.Integer, db.ForeignKey(
        'usuarios.id'), nullable=False)
    cliente = db.Column(db.String(255), nullable=True)
    data_realizacao = db.Column(db.DateTime, nullable=False)
    limpeza = db.Column(db.Boolean, nullable=True, default=False)
    poda_plantas = db.Column(db.Boolean, nullable=True, default=False)
    poda_grama = db.Column(db.Boolean, nullable=True, default=False)
    outros = db.Column(db.Boolean, nullable=True, default=False)
    outros_descricao = db.Column(db.String(255), nullable=True)
    responsavel_type = db.Column(db.String(50), nullable=False)
    assinatura_jardineiro = db.Column(db.String(255), nullable=True)
    assinatura_responsavel = db.Column(db.String(255), nullable=True)
    observacao = db.Column(db.String(255), nullable=True)
    data = db.Column(db.DateTime, nullable=False)

    def __init__(self, cliente_id, usuario_id, cliente, data_realizacao, responsavel_type, assinatura_responsavel, assinatura_jardineiro, limpeza=False, poda_plantas=False, poda_grama=False, outros=False, outros_descricao=None, observacao=None, data=None):
        self.cliente_id = cliente_id
        self.usuario_id = usuario_id
        self.cliente = cliente
        self.data_realizacao = data_realizacao
        self.responsavel_type = responsavel_type
        self.assinatura_responsavel = assinatura_responsavel
        self.assinatura_jardineiro = assinatura_jardineiro
        self.limpeza = limpeza
        self.poda_plantas = poda_plantas
        self.poda_grama = poda_grama
        self.outros = outros
        self.outros_descricao = outros_descricao
        self.observacao = observacao
        self.data = data

File: static/js/OrdemServico.js


//Abilitar e desabilitar campo Serviço
    function toggleOutrosDescricao() {
        var outrosCheckbox = document.getElementById('outrosServ');
        var outrosDescricao = document.getElementById('outrosServdescricao');

        // Se a opção "Outros" estiver marcada, mostra o campo de descrição, caso contrário, oculta
        if (outrosCheckbox.checked) {
            outrosDescricao.style.display = 'block';
        } else {
            outrosDescricao.style.display = 'none';
        }
    }

    // Adiciona um listener para o evento 'change' no checkbox de outros
    document.getElementById('outrosServ').addEventListener('change', toggleOutrosDescricao);

    ///////////////////////////////
    // Aqui onde eu populo os meus clientes
    function checkOtherOption() {
        var select = document.getElementById("cliente_id");
        var outroInput = document.getElementById("NomeCliente");

        if (select.value === "outroCliente") { // Se a opção "Outros " for selecionada
            outroInput.style.display = "block"; // Mostrar o campo de texto
        } else {
            outroInput.style.display = "none"; // Esconder o campo de texto
            outroInput.value = " "
        }
    }


    // Aqui e onde vai conter a funçãoAssinatura
    var canvasJardim = document.getElementById('canvasAssinatura');
    var canvasRespo = document.getElementById('canvasAssinaturaResp');
    var spadJardim = new SignaturePad(canvasJardim);
    var SpadRespo = new SignaturePad(canvasRespo);

    //Assianatura Jardineiro
    var campoAssinatura = document.getElementById('assinturaJardineiro');
    campoAssinatura.addEventListener('click', function() {
        canvasJardim.style.display = 'block';
    });

    //Assinatura Responsavel
    var campoAssinaturaResp = document.getElementById('assinturaResponsavel');
    campoAssinaturaResp.addEventListener('click', function() {
        SpadRespo.style.display = 'block';
    });


// <!-- Função para validações -->


    function validarFormularioSer() {

        // Verificar se pelo menos um tipo de serviço foi selecionado
        var limpeza = document.getElementById("limpeza").checked;
        var podaGrama = document.getElementById("poda-grama").checked;
        var podaPlantas = document.getElementById("poda-plantas").checked;
        var outros = document.getElementById("outrosServ").checked;
        var outrosServdescricao = document.getElementById("outrosServdescricao");

        if (!limpeza && !podaGrama && !podaPlantas && !outros) {
            alert("Selecione pelo menos um tipo de serviço!");
            return false; // Impede o envio do formulário
        }
        if (outros) {
            if (outrosServdescricao != "")
                alert("Descreva o outro Serviço!");
                 return false; // Impede o envio do formulário
        }


        
        //Verificar se outros clientes esta preenchido
        var select = document.getElementById("clientes");
        var outroInput = document.getElementById("outroCliente");

        if (select.value === "outros") { // Se a opção "Outros " for selecionada
            if (outrosClientes =="")
                alert("Selecione pelo menos um tipo de Cliente!");
                return false
          }

        // Outras validações, se necessário
        return true; // Permite o envio do formulário se todas as validações passarem
        }



        
    function VoltarOS() {
        // Limpe os campos do formulário
        document.getElementById('cadOrdemServicoForm').reset();
        window.location.href='ordem_servico'
    }
    
      function validarFormulario() {
        // Verifique se todos os campos obrigatórios estão preenchidos
        var clienteId = document.getElementById('cliente_id').value;
        var dataRealizacao = document.getElementById('dataRealizacao').value;
        var assinaturaJardineiro = document.getElementById('assinturaJardineiro').value;
        var responsavel = document.getElementById('responsavel').value;

        if (!clienteId || !dataRealizacao || !assinaturaJardineiro || !responsavel) {
            // Se algum campo estiver vazio, exiba uma mensagem de erro
            alert('Por favor, preencha todos os campos obrigatórios.');
            return false; // Impede o envio do formulário
        }

        // Se todos os campos obrigatórios estiverem preenchidos, retorne true para permitir o envio do formulário
        return true;
    }